proversity_reports_script/report_backend/enrollment_report.py

```diff
@@ -0,0 +1,284 @@
+                'Program_of_Interest': self.extra_data.get('PROGRAM_OF_INTEREST', ''),
+                'Intake_of_Intent': user.get('intake_of_intent', ''),
+                'Lead_Source': self.extra_data.get('LEAD_SOURCE', DEFAULT_LEAD_SOURCE),
+                'Program_Code': course,
+                'Drupal_ID': user.get('user_id', ''),
+            }
+
+            batch.append(user_salesforce_data)
+
+            if len(batch) == BATCH_SIZE or index == len(course_data):
+                salesforce_data.append(batch)
+                batch = []
+
+        return salesforce_data
+
+    def _create_saleforce_enrollments(self, enrollment_data):
+        """
+        Creates the enrollment list into Salesforce.
+
+        Args:
+            enrollment_data: List of list with the enrollment data.
+        Returns:
+            salesforce_result: List wiht the contact id data from Salesforce.
+        """
+        request_session = requests.Session()
+        salesforce_details = self._get_salesforce_details()
+        enrollment_api_url = self.extra_data.get('SALESFORCE', {}).get(
+            'SALESFORCE_API_ENROLLMENT_URL',
+            ''
+        )
+
+        if not enrollment_api_url:
+            print('SALESFORCE_API_ENROLLMENT_URL was not provided.')
+            exit()
+
+        api_url = '{}{}'.format(
+            salesforce_details.get('instance_url', ''),
+            enrollment_api_url,
+        )
+
+        request_session.headers.update({
+            'Authorization': '{} {}'.format(
+                salesforce_details.get('token_type', ''),
+                salesforce_details.get('access_token', ''),
+            ),
+        })
+
+        print('Creating enrollments on Salesforce...')
+
+        salesforce_result = []
+
+        for enrollments in enrollment_data:
+            request_data = {
+                'enrollments': enrollments,
+            }
+
+            api_response = request_session.post(api_url, json=request_data)
+
+            if api_response.ok:
+                salesforce_result.append(api_response.json())
+            else:
+                print('There was an error calling the Salesforce API: ', api_response.text)
+
+        return salesforce_result
+
+    def _get_salesforce_details(self):
+        """
+        Returns the Salesforce details from the authentication request.
+
+        Returns:
+            A dict with some Salesforce deatils.
+            {
+                'access_token': Salesforce access token.
+                'instance_url': Salesforce instance url.
+                'token_type': Access token type.
+            }
+        """
+        print('Getting Salesforce token deatils...')
+
+        salesforce_data = self.extra_data.get('SALESFORCE', {})
+
+        if not salesforce_data:
+            print('SALESFORCE data was not provided.')
+            exit()
+
+        salesforce_url = salesforce_data.get('AUTHENTICATION_URL', '')
```

Comment:
we should validate this data before use it

Reply:
Addressed.